shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-common/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java

```diff
@@ -124,7 +126,16 @@ private int getLastPartition(final Range<Comparable<?>> valueRange) {
     }
     
     private long parseDate(final Comparable<?> shardingValue) {
-        LocalDateTime dateValue = LocalDateTime.parse(shardingValue.toString(), DATE_TIME_FORMAT);
+        LocalDateTime dateValue;
```

Comment:
Thank you very much for your contribution. 

Now I would like to know when shardingValue is Date or LocalDateTime?

We can't judge the treatment so simply,  because we define DateTimeFormatter like this 

```
DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
```

if you data or config not follow this, An exception is thrown, so you need to follow this DateTimeFormatter .

Reply:
It happened when the type of the sharding value is LocalDateTime. Here is the configuration:

```
          dt_partition_detection:
            actual-data-nodes: dsd$->{0..9}.dt_partition_detection_$->{2013..2025}0$->{1..4}
            database-strategy:
              standard:
                sharding-column: sys_org_code
                sharding-algorithmName: detection-db-sharding
            tableStrategy:
              standard:
                sharding-column: time
                sharding-algorithm-name: detection-table-sharding
            key-generate-strategy:
              type: SNOWFLAKE
              column: id

...
            detection-table-sharding:
              type: INTERVAL
              props:
                datetime-pattern: yyyy-MM-dd HH:mm:ss
                datetime-lower: '2013-01-01 00:00:00'
                sharding-suffix-pattern: yyyyQQ
                datetime-interval-amount: '3'
                datetime-interval-unit: MONTHS
```

Using INTERVAL sharding rule，when insert a entity, whose field **time** is LocalDateTime. 

After PreparedStatement.execute(); it will call doSharding using the arguments passed to PreparedStatement.

**time** is not called with the toString() method, but directly pass the raw value. Like the image showing:

![image](https://user-images.githubusercontent.com/1018698/102947853-4a5ead00-44ff-11eb-8953-dd9170f069cc.png)

Reply:
can you config  datetime-pattern: yyyy-MM-ddTHH:mm:ss ?

Reply:
Thanks.

After I configured the config as you said, it is normal, but the value passed to doSharding is still LocalDateTime, The configuration just make the formatter yyyy-MM-ddTHH:mm:ss the same as LocalDateTime.toString() using. So the different configuaration is not respected.

![image](https://user-images.githubusercontent.com/1018698/102956523-4c326b80-4513-11eb-97e0-19c290ae514e.png)

Reply:
so this pr ,we can not merged, Do you have any other fix,  if not can you close this pr?